src/components/Profile/Profile.jsx

```diff
@@ -1,20 +1,47 @@
 import * as React from "react"
-import { useState, useEffect } from "react"
+import { useState, useEffect, useRef } from "react"
 import { Link, useNavigate } from "react-router-dom"
-import { Card, Button } from "react-bootstrap"
+import { Card, Button, Badge } from "react-bootstrap"
+import { Typeahead } from 'react-bootstrap-typeahead'
+import { doc, updateDoc } from "firebase/firestore"
 import { useAuth } from "../../contexts/AuthContext"
 import { db } from "../../firebase"
 import "./Profile.css"
+import 'react-bootstrap-typeahead/css/Typeahead.css'
+import CustomTooltip from "../CustomTooltip/CustomTooltip"
 
 export default function Profile(props) {
   const { logout, currentUser } = useAuth()
+  const typeaheadDietsRef = useRef(null);
+  const typeaheadAllergiesRef = useRef(null);
   const [userProducts, setUserProducts] = useState([])
+  const [userCart, setUserCart] = useState([])
+  const [userPrimDiet, setUserPrimDiet] = useState("")
+  const [userDiets, setUserDiets] = useState([])
+  const [userAllergies, setUserAllergies] = useState([])
+  const [error, setError] = useState("")
+  const [showUserPrimDietForm, setShowUserPrimDietForm] = useState(false)
+  const primDiets = ["Gluten Free", "Ketogenic", "Vegetarian", "Vegan", "Pescetarian", "Paleo"]
+  const [primDietChecked, setPrimDietChecked] = useState([])
+  const [showUserDietsForm, setShowUserDietsForm] = useState(false)
+  const [diets, setDiets] = useState(["Gluten Free", "Ketogenic", "Vegetarian", "Vegan", "Pescetarian", "Paleo"])
+  const [dietsChecked, setDietsChecked] = useState([])
+  const [showUserAllergiesForm, setShowUserAllergiesForm] = useState(false)
+  const allergies = ["Diary", "Peanut", "Soy", "Egg", "Shellfish", "Tree Nut", "Gluten"]
+  const [allergiesChecked, setAllergiesChecked] = useState([])
   const navigate = useNavigate()
 
+  const primDietDialogue = "Primary dietary restriction will filter out recipes that don't follow this restriction"
+  const dietDialogue = "Secondary dietary restrictions will not filter out recipes that don't follow this restriction, but will show a warning at the top of recipe instructions"
+
   useEffect(() => {
```

Comment:
I don't think this needs to be in a useEffect (https://reactjs.org/docs/hooks-effect.html). You can simply just read set the default value of your state based on the props. Example: const [userProducts, setUserProducts] = useState(props.isLoading? [] : userInfo.data.products). With the useEffect you are causing an extra render, but much more importantly making the code more complex and harder to reason about.